v2/devdoc/log_context_property_type_if_requirements.md

```diff
@@ -63,6 +71,22 @@ typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
 
 `LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE` returns the implemented type as known by `c_logging`.
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(type_name)
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(type_name)
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_IF_01_001: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(type_name)` shall produce the token `{type_name}_log_context_property_type`. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_INIT(type_name)
+
+`LOG_CONTEXT_PROPERTY_TYPE_INIT(type_name)` initializes a property from some source value arguments.
+
+For example for basic types the initialization consists of simply copying the bytes that make up the value to the destination address.
+
+For more complex data types (like a reference counted value), this could mean storing the reference and incrementing the reference count.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_IF_01_002: [** `LOG_CONTEXT_PROPERTY_TYPE_INIT(type_name)` shall produce the token `{type_name}_log_context_property_type_init`. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(type_name)
+
+`LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(type_name)` returns the amount of data needed for initializing a property by using `LOG_CONTEXT_PROPERTY_TYPE_INIT(type_name)`.
```

Comment:
>LOG_CONTEXT_PROPERTY_TYPE_INIT [](http://example.com/codeflow?start=130&length=30)

ascii_char_ptr has extra arguments here, "format" and "...".

Is this the same interface or strings are "special"?
```
int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(ascii_char_ptr)(const char* format, ...);
```